### PR TITLE
[Port to 2.0] Fix failing tests on OSX 2.0 branch.

### LINKED ISSUE #22089: Test: System.Net.Http.Functional.Tests.HttpClientHandler_DangerousAcceptAllCertificatesValidator_Test/SetDelegate_ConnectionSucceeds failed with "System.Net.Http.CurlException : SSL connect error"
Opened on behalf of @Jiayili1

The test `System.Net.Http.Functional.Tests.HttpClientHandler_DangerousAcceptAllCertificatesValidator_Test/SetDelegate_ConnectionSucceeds(acceptedProtocol: Tls, requestOnlyThisProtocol: False)` has failed.

System.Net.Http.HttpRequestException : An error occurred while sending the request.
    ---- System.Net.Http.CurlException : SSL connect error

        Stack Trace:

           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
       at System.Net.Http.HttpClient.<FinishSendAsyncBuffered>d__58.MoveNext() in /Users/buildagent/agent/_work/41/s/corefx/src/System.Net.Http/src/System/Net/Http/HttpClient.cs:line 462
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at System.Net.Http.Functional.Tests.HttpClientHandler_DangerousAcceptAllCertificatesValidator_Test.<>c__DisplayClass1_0.<<SetDelegate_ConnectionSucceeds>b__0>d.MoveNext() in /Users/buildagent/agent/_work/41/s/corefx/src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.AcceptAllCerts.cs:line 46
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at System.Net.Test.Common.LoopbackServer.<>c__DisplayClass3_0.<CreateServerAsync>b__0(Task t) in /Users/buildagent/agent/_work/41/s/corefx/src/Common/tests/System/Net/Http/LoopbackServer.cs:line 57
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
       at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot)
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at System.Net.Http.Functional.Tests.HttpClientHandler_DangerousAcceptAllCertificatesValidator_Test.<SetDelegate_ConnectionSucceeds>d__1.MoveNext() in /Users/buildagent/agent/_work/41/s/corefx/src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.AcceptAllCerts.cs:line 44
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
    ----- Inner Stack Trace -----
       at System.Net.Http.CurlHandler.ThrowIfCURLEError(CURLcode error) in /Users/buildagent/agent/_work/41/s/corefx/src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs:line 644
       at System.Net.Http.CurlHandler.MultiAgent.FinishRequest(StrongToWeakReference`1 easyWrapper, CURLcode messageResult) in /Users/buildagent/agent/_work/41/s/corefx/src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs:line 869
Build : 2.0.0 - 20170711.01 (Core Tests)
Failing configurations:
- OSX.1013.Amd64-x64
  - Debug
  - Release

Detail: https://mc.dot.net/#/product/netcore/200/source/official~2Fcorefx~2Frelease~2F2.0.0~2F/type/test~2Ffunctional~2Fcli~2F/build/20170711.01/workItem/System.Net.Http.Functional.Tests/analysis/xunit/System.Net.Http.Functional.Tests.HttpClientHandler_DangerousAcceptAllCertificatesValidator_Test~2FSetDelegate_ConnectionSucceeds(acceptedProtocol:%20Tls,%20requestOnlyThisProtocol:%20False)

### PR DESCRIPTION
**Pull request by Priya91**
cc @Petermarcu @bartonjs 

fixes #22089 #22090

### CODE REVIEW

**Comment by Petermarcu** [comment]
Did this fix it in Master?

**Comment by Priya91** [comment]
> Did this fix it in Master?

The latest mission control for master, doesn't have the test failures https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Fcli~2F/build/20170713.02

File: src/System.Net.HttpListener/tests/System.Net.HttpListener.Tests.csproj [hunk 1]
```diff
@@ -34,5 +34,8 @@
     </Compile>
     <Compile Include="HttpListenerTimeoutManagerTests.cs" />
   </ItemGroup>
+  <ItemGroup Condition="'$(TargetsOSX)'=='true'">
+    <TestCommandLines Include="ulimit -n 70000" />
```
Comment (bartonjs) [inline src/System.Net.HttpListener/tests/System.Net.HttpListener.Tests.csproj]: When I tried this on Tomas' machine it told me 70000 was invalid.  But it's worth trying, I guess.

Comment (Priya91) [inline src/System.Net.HttpListener/tests/System.Net.HttpListener.Tests.csproj]: oh ok, let me reduce it to 7000..

**Comment by Priya91** [comment]
@bartonjs Crypto tests are failing on osx10.12 with
```
MESSAGE:
Interop+AppleCrypto+AppleCommonCryptoCryptographicException : A default keychain could not be found.
+++++++++++++++++++
STACK TRACE:
at Interop.AppleCrypto.SecKeychainCopyDefault() in /Users/dotnet-bot/j/workspace/dotnet_corefx/release_2.0.0/osx10.12_debug_prtest/src/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.Keychain.cs:line 108 at Internal.Cryptography.Pal.StorePal.AppleKeychainStore.OpenDefaultKeychain(OpenFlags openFlags) in /Users/dotnet-bot/j/workspace/dotnet_corefx/release_2.0.0/osx10.12_debug_prtest/src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/StorePal.AppleKeychainStore.cs:line 83 at Internal.Cryptography.Pal.StorePal.FromSystemStore(String storeName, StoreLocation storeLocation, OpenFlags openFlags) in /Users/dotnet-bot/j/workspace/dotnet_corefx/release_2.0.0/osx10.12_debug_prtest/src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/StorePal.cs:line 106 at System.Security.Cryptography.X509Certificates.X509Store.Open(OpenFlags flags) in /Users/dotnet-bot/j/workspace/dotnet_corefx/release_2.0.0/osx10.12_debug_prtest/src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509Store.cs:line 128 at System.Security.Cryptography.X509Certificates.Tests.X509StoreTests.Dispose_IsOpenFalse() in /Users/dotnet-bot/j/workspace/dotnet_corefx/release_2.0.0/osx10.12_debug_prtest/src/System.Security.Cryptography.X509Certificates/tests/X509StoreTests.cs:line 191
```

```
System.Security.Cryptography.X509Certificates.Tests.X509StoreTests.TestDispose
System.Security.Cryptography.X509Certificates.Tests.X509StoreTests.Constructor_OpenFlags_OpenAnyway
System.Security.Cryptography.X509Certificates.Tests.X509StoreTests.Dispose_IsOpenFalse
System.Security.Cryptography.X509Certificates.Tests.X509StoreTests.ReOpen_IsOpenTrue
System.Security.Cryptography.X509Certificates.Tests.X509StoreTests.AddDisposedThrowsCryptographicException
System.Security.Cryptography.X509Certificates.Tests.X509StoreTests.Constructor_StoreHandle
System.Security.Cryptography.X509Certificates.Tests.X509StoreTests.Constructor_OpenFlags
System.Security.Cryptography.X509Certificates.Tests.X509StoreTests.AddReadOnlyThrows
System.Security.Cryptography.X509Certificates.Tests.X509StoreTests.ReadMyCertificates
System.Security.Cryptography.X509Certificates.Tests.X509StoreTests.RemoveReadOnlyThrowsWhenFound
System.Security.Cryptography.X509Certificates.Tests.X509StoreTests.Open_IsOpenTrue
System.Security.Cryptography.X509Certificates.Tests.X509StoreTests.OpenMyStore
System.Security.Cryptography.X509Certificates.Tests.X509StoreTests.AddReadOnlyThrowsWhenCertificateExists
System.Security.Cryptography.X509Certificates.Tests.X509StoreTests.Constructor_OpenFlags_StoreName
System.Security.Cryptography.X509Certificates.Tests.CertTests.X509Cert2ToStringVerbose
```

**Comment by bartonjs** [comment]
@mmitche Please take the converted OSX machine out of rotation.

**Comment by bartonjs** [comment]
@dotnet-bot Test Innerloop OSX10.12 Debug x64 Build and Test please

**Comment by mmitche** [comment]
@bartonjs Do you have some time to look into why this is failing early next week?  I'd like to convert the rest of the machines (since it's more reliable, and not running as root)

**Comment by bartonjs** [comment]
> @bartonjs Do you have some time to look into why this is failing early next week?

The tests are failing because [SecKeychainCopyDefault()](https://developer.apple.com/documentation/security/1400743-seckeychaincopydefault?language=objc) is failing.  I don't know what causes that (other than the user not having a keychain, which isn't helpful... because what I don't know is why \[that\]).

I speculate that this means "the user has never logged on via the GUI".  But that's entirely psychic debugging.

**Comment by mmitche** [comment]
The user definitely has logged in (it’s the dotnet-bot user)

**Pull request merged**